src/utils/perfagg.py

```diff
@@ -86,6 +87,133 @@
 }
 
 
+# joins disparate runs less dumbly than rocprof
+def join_prof(workload_dir, join_type, out=None):
+    # Set default output directory if not specified
+    if out == None:
+        out = workload_dir + "/pmc_perf.csv"
+    files = glob.glob(workload_dir + "/" + "pmc_perf_*.csv")
+    df = None
+
+    for i, file in enumerate(files):
+        # _df = parse_rocprof_kernels(file)
+
+        _df = pd.read_csv(file)
+        key = _df.groupby("KernelName").cumcount()
```

Comment:
I actually thought of something overnight -- when we are trying to join via 'grid', we need to actually generate the key based on the groupby(("KernelName", "grd")), such that we count unique dispatches of each kernel + grid combination.

I think that actually just means we need to move the `join_type` up to this line, to do like:

```
if join_type == "kernel":
  key = _df.groupby("KernelName").cumcount()
elif join_type == "grid":
  _df.groupby(("KernelName", "grd")).cumcount()
else:
  <error>
```

Then I think we just continue as below

Reply:
https://github.com/AMDResearch/omniperf/blob/44edef4635c18ee3f2bb764f7078db05dfbbecd6/src/utils/perfagg.py#L104-L111